src/api/app/models/bs_request.rb

```diff
@@ -294,22 +294,28 @@ def is_reviewer? (user)
     false
   end
 
-  def remove_reviews(opts)
+  def obsolete_reviews(opts)
     return false unless opts[:by_user] or opts[:by_group] or opts[:by_project] or opts[:by_package]
     reviews.each do |review|
       if review.by_user and review.by_user == opts[:by_user] or
           review.by_group and review.by_group == opts[:by_group] or
           review.by_project and review.by_project == opts[:by_project] or
           review.by_package and review.by_package == opts[:by_package]
-        logger.debug "Removing review #{review.id}"
-        self.reviews.delete(review)
-        if self.reviews.none?
+        logger.debug "Obsoleting review #{review.id}"
+        review.state = :obsoleted
```

Comment:
we can change this in all places in another commit, but I like to keep this consistent with rest for now.